Modules/Core/Common/include/itkMacro.h

```diff
@@ -435,11 +435,13 @@ namespace itk
 
 
 // Internal macro (not part of the public ITK API), used to implement `GetNameOfClass()` member functions.
-#define itkInternalGetNameOfClassImplementationMacro(thisClass)                                              \
-  {                                                                                                          \
-    static_assert(std::is_same_v<thisClass, std::remove_const_t<std::remove_reference_t<decltype(*this)>>>); \
-    return #thisClass;                                                                                       \
-  }                                                                                                          \
+#define itkInternalGetNameOfClassImplementationMacro(thisClass)                                             \
+  {                                                                                                         \
+    static_assert(std::is_same_v<thisClass, std::remove_const_t<std::remove_reference_t<decltype(*this)>>>, \
+                  "The macro argument `" #thisClass                                                         \
```

Comment:
Useful as is. It would be more useful if we added after "name of this class" its name (`std::remove_const_t<std::remove_reference_t<decltype(*this)>>`). Can this be accomplished?

Reply:
Thanks for your comment @dzenanz. As far as I can see, with C++17, the (optional) message of a static_assert must be a simple literal string, so it cannot have something like `type(*this).name()`. https://en.cppreference.com/w/cpp/language/static_assert  But maybe with C++26... 😸  

Related C++ proposal: [user-generated `static_assert` messages](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2741r3.pdf) by Corentin Jabot